aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -206,6 +328,67 @@ extern "C" void strsm_(char *side, char *uplo, char *trans, char *diag, int *n,
 namespace at {
 namespace meta {
 
+TORCH_META_FUNC(linalg_ldl_factor_ex)
+(const Tensor& self, bool upper, bool hermitian, bool check_errors) {
+  at::native::squareCheckInputs(self, "torch.linalg.ldl_factor_ex");
+  at::native::checkFloatingOrComplex(self, "torch.linalg.ldl_factor_ex");
+
+  auto ndim = self.dim();
+  auto n = self.size(-2);
+
+  // prefer column major strides
+  auto ldu_strides = at::detail::defaultStrides(self.sizes());
+  ldu_strides[ndim - 2] = 1;
+  ldu_strides[ndim - 1] = std::max<int64_t>(1, n);
+  set_output(0, self.sizes(), ldu_strides, self.options(), {}); // factors
+
+  auto pivots_shape =
+      IntArrayRef(self.sizes().data(), ndim - 1); // self.shape[:-1]
+  set_output(
+      1, pivots_shape, {}, self.options().dtype(ScalarType::Int), {}); // pivots
+
+  auto info_shape =
+      IntArrayRef(self.sizes().data(), ndim - 2); // self.shape[:-2]
+  set_output(
+      2, info_shape, {}, self.options().dtype(ScalarType::Int), {}); // info
+}
+
+TORCH_META_FUNC(linalg_ldl_solve)
+(const Tensor& factors,
+ const Tensor& pivots,
+ const Tensor& B,
+ bool upper,
+ bool hermitian) {
+  at::native::squareCheckInputs(factors, "torch.linalg.ldl_solve");
+  at::native::checkFloatingOrComplex(factors, "torch.linalg.ldl_solve");
+  at::native::linearSolveCheckInputs(B, factors, "torch.linalg.ldl_solve");
+  TORCH_CHECK(
+      B.dim() >= 2,
+      "torch.linalg.ldl_solve: Expected B to have at least 2 dimensions, but it has ",
+      B.dim(),
+      " dimensions instead");
+  TORCH_CHECK(
+      at::isIntegralType(pivots.scalar_type(), /*includeBool=*/false),
```

Comment:
No, LAPACK requires kInt because we use 32-bit interface, cuSOLVER requires 64-bit (kLong), so we'll allow any integer that will be converted either to kInt or kLong.
https://github.com/pytorch/pytorch/pull/69828#discussion_r786428218.